src/utils/HexStringToByteArray.js

```diff
@@ -1,13 +1,13 @@
+// TODO consider using Buffer.from(<string>, 'hex') instead
 const HexStringToByteArray = function (str) {
-    const start = str.startsWith('0x') ? 2 : 0
-    let ints = [];
-
-    for (let i = start, charsLength = str.length; i < charsLength; i += 2) {
-        const hex = str.substring(i, i + 2)
-        ints.push(parseInt(hex, 16));
+    const match = str.match(new RegExp(/^(0x)?([a-f0-9]*)$/, 'i'))
```

Comment:
I'm using `*` instead of `+` in regexp to be consistent with `Buffer.from`

Reply:
makes sense, `Buffer.from` seems to be NodeJS implementation only, while this lib aims for VanillaJS in browser support .